### PR TITLE
DDP-5505 support text question institution suggestion type

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

Implements server-side of DDP-5505. We add the `INSTITUTION` suggestion type for `TEXT` questions. Most of the heavy-lifting of autocompletion in on client-side in https://github.com/broadinstitute/ddp-angular/pull/404. I also went ahead and added support for taking `limit` query param in the Institution Suggestion API.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 1]
```diff
@@ -1470,8 +1470,9 @@ Question.Text:
         enum:
           - NONE
           - DRUG
-          - INCLUDED
           - CANCER
+          - INCLUDED
+          - INSTITUTION
```
Comment (yufengwng) [inline pepper-apis/docs/specification/src/components/schemas.yml:1475]: @fat23cat @bskinner Here are the API changes.

File: pepper-apis/docs/specification/src/pepper.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ openapi: 3.0.0
 servers: []
 info:
   description: 'Pepper API specification'
-  version: 1.9.5
+  version: 1.9.8
```
Comment (yufengwng) [inline pepper-apis/docs/specification/src/pepper.yml:5]: I think next release is going to be `v1.9.8`, so bumping this up.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/route/GetInstitutionSuggestionsRoute.java [hunk 3]
```diff
@@ -52,14 +52,25 @@
...
         DDPAuth ddpAuth = RouteUtil.getDDPAuth(request);
         String operator = ddpAuth.getOperator();
-        Map<String, InstitutionSuggestionDto> suggestions = getSuggestionDtos(namePattern, LIMIT);
+        Map<String, InstitutionSuggestionDto> suggestions = getSuggestionDtos(namePattern, limit);
         LOG.info(
-                "Sent {} suggestions matching the pattern {} for operator {}",
+                "Sent {} suggestions matching the pattern '{}' for operator {}",
                 suggestions.size(), namePattern, operator
         );
-        return new ArrayList(suggestions.values());
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/route/GetInstitutionSuggestionsRoute.java:62]: This is technically a bug. `suggestions` is a list of `InstitutionSuggestionDto` objects while the return type is list of `InstitutionSuggestion` objects. It just so happens that those two objects have the same shape/properties so the JSON ended up the same, but we should explicitly do the conversion.

**Review by ssettipalli — APPROVED** [review]
looks good to me

**Pull request merged**